validator/src/main/java/com/google/daq/mqtt/validator/Validator.java

```diff
@@ -473,7 +484,7 @@ private File makeDeviceDir(String deviceId) {
     return deviceDir;
   }
 
-  private String makeSchemaName(Map<String, String> attributes) {
+  private String mungeAndExtractSchema(Map<String, String> attributes) {
```

Comment:
It isn't immediately obvious why we are crossing subFolder and subType values. 

Can we either make it more clear with comments, or, reference the doc? envelope.md doesn't explain it ... ?

Reply:
Ok, you shamed me into fixing the underlying confusion, so the crossing the values is no longer required. It did involve some additional changes, but it's now a lot cleaner overall, I think.